ACM-12171: Add whats new and deprecated info (#6584)
open-cluster-management/rhacm-docs: clusters/release_notes/whats_new.adoc

[#hosted-control-plane]
== Hosted control planes

> * The `--aws-creds` flag in the `hcp` command line interface (CLI) is deprecated. Now, you need to use the `--sts-creds` and `--role-arn` flags in the `hcp` CLI. To use the `--sts-creds` and `--role-arn` flags, you must create an AWS IAM role and STS credentials. For more information, see xref:../hosted_control_planes/create_role_sts_aws.adoc#create-role-sts-aws[Creating an AWS IAM role and STS credentials].

Reviewer: We do not announce deprecations in what's new, it's only for new features.

Author: Ack. Rephrased

> * Now, you need to use the `--sts-creds` and `--role-arn` flags instead of the `--aws-creds` flag in the `hcp` command line interface. To use the `--sts-creds` and `--role-arn` flags, you must create an Amazon Web Services (AWS) Identity and Access Management (IAM) role and Security Token Service (STS) credentials. For more information, see xref:../hosted_control_planes/create_role_sts_aws.adoc#create-role-sts-aws[Creating an AWS IAM role and STS credentials].

Reviewer: ```suggestion
* The `--sts-creds` and `--role-arn` flags replace the deprecated `--aws-creds` flag in the `hcp` command line interface. Create an Amazon Web Services (AWS) Identity and Access Management (IAM) role and Security Token Service (STS) credentials to use the `--sts-creds` and `--role-arn` flags. For more information, see xref:../hosted_control_planes/create_role_sts_aws.adoc#create-role-sts-aws[Creating an AWS IAM role and STS credentials].
```

What's new generally needs to be positive, reworded with Brandi's help

Author: Thanks. Addressed.
